### DISCUSSION TITLE
How do I update Grocy?

### DISCUSSION THREAD

**Discussion by jlafuenti** [hassio-addons/addon-grocy#449] [Q&A]
Stupid question, but I haven't been able to find it on google...
My grocy instance that is from the Add on is running version 4.0.3
However Grocy in github is running version 4.2.0
How do I get the add on to pull the latest version?

**Comment by dwildstr** [comment]
There's no easy way to update from HA itself because there are a couple of layers in between --- it's the docker configuration in the addon itself which specifies version 4.0.3 (specifically, line 9 of `grocy/Dockerfile`), and Home Assistant can only update addons (which are self-contained docker containers), not the software within the addons. The latest stable release of addon-grocy is v.0.21.0, from January, which doesn't implement the updated grocy version. The current codebase for this addon _does_ use grocy 4.2.0 and has since March (see #434), but until the next official release of addon-grocy, the official Home Assistant add-on won't implement that change.

If you're willing to take a step into the unknown, there _is_ an addon repository which implements the development versions of addons, so it uses grocy 4.2.0. You can find more details on this repository [here](https://addons.community/edge/). Because the edge version of grocy is technically a separate addon (with a different docker ID) rather than an update to the stable version, it's got a separate docker container and you'd need to copy all your configuration and data over. Also, because it's a development version, there could theoretically be all sorts of broken functionality in it (in practice there shouldn't be much, because addon-grocy is just pulling in released software, even in its development version, but it could have mismatched versions on its dependencies or somesuch).

For my part, that seems like a hassle and I'd just as soon wait for the official release, which I hope will be soon, but if you really want to get to 4.2.0 now, there are options.
